src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicTypeInferenceService.TypeInferrer.vb

```diff
@@ -801,14 +803,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Return SpecializedCollections.SingletonEnumerable(Me.Compilation.GetSpecialType(SpecialType.System_Boolean))
             End Function
 
-            Private Function InferTypeInMemberAccessExpression(expression As MemberAccessExpressionSyntax) As IEnumerable(Of ITypeSymbol)
+            Private Function InferTypeInMemberAccessExpression(expression As MemberAccessExpressionSyntax, Optional previousToken As SyntaxToken = Nothing) As IEnumerable(Of ITypeSymbol)
                 Dim awaitExpression = expression.GetAncestor(Of AwaitExpressionSyntax)
                 Dim lambdaExpression = expression.GetAncestor(Of LambdaExpressionSyntax)
                 If Not awaitExpression?.Contains(lambdaExpression) AndAlso awaitExpression IsNot Nothing Then
                     Return InferTypes(awaitExpression.Expression)
                 End If
 
-                Return SpecializedCollections.EmptyEnumerable(Of ITypeSymbol)()
+                ' If we're position based, walk up the chain of member accesses
+                ' and infer the type at that position
+                If previousToken.IsKind(SyntaxKind.DotToken) Then
+                    If expression.IsParentKind(SyntaxKind.SimpleMemberAccessExpression) Then
+                        Dim parent = DirectCast(expression.Parent, MemberAccessExpressionSyntax)
+                        Return InferTypeInMemberAccessExpression(parent, parent.OperatorToken)
+                    End If
+
+                    Return InferTypes(expression.SpanStart)
```

Comment:
The if at 816 looks at the Kind of the `parent` of the expression that had the dot token. It's trying to walk through all the member accesses until we find the position of the place that expects an expression.